WEBPACK.md

```diff
@@ -0,0 +1,144 @@
+
+```javascript
+// Include necessary modules
+const webpack = require('webpack');
+const glob = require('glob');
+const path = require('path');
+const MergeIntoSingleFilePlugin = require('webpack-merge-and-include-globally');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+
+const config = {
+  // "production" or "development", set by cross-env in NPM scripts
+  // Determines which optimizations it needs to apply to our JavaScript files
+  mode: process.env.NODE_ENV,
+  // Files to bundle(in order from the top to bottom)
+  entry: [
+    /** glob.sync(folder) returns an array of file paths in the folder
+     * ...(spread) operator spreads out an array
+     * ["f1", "f2"] -> "f1", "f2"
+     * entry only accepts an array of strings(files) to bundle or a single entry file
+     */
+    ...glob.sync('./src/util/*'),
+    './src/DistortableImageOverlay.js',
+    './src/DistortableCollection.js',
+    './src/edit/getEXIFdata.js',
+    './src/edit/handles/EditHandle.js',
+    ...glob.sync('./src/edit/handles/*', {
+      ignore: './src/edit/handles/EditHandle.js'
+    }),
+    './src/iconsets/IconSet.js',
+    './src/iconsets/KeymapperIconSet.js',
+    './src/iconsets/ToolbarIconSet.js',
+    './src/edit/actions/EditAction.js',
+    ...glob.sync('./src/edit/actions/*', {
+      ignore: './src/edit/actions/EditAction.js'
+    }),
+    './src/edit/toolbars/DistortableImage.PopupBar.js',
+    './src/edit/toolbars/DistortableImage.ControlBar.js',
+    './src/edit/DistortableImage.Edit.js',
+    './src/edit/DistortableCollection.Edit.js',
+    './src/components/DistortableImage.Keymapper.js',
+    './src/mapmixins/DoubleClickZoom.js',
+    ...glob.sync('./src/mapmixins/*', {
+      ignore: './src/mapmixins/DoubleClickZoom.js'
+    })
+  ],
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    filename: 'leaflet.distortableimage.js',
+    /**
+     * Webpack live server, on change in the source code, compiles files again and 	     * generates new hot update files. Instead of generating new files on each change,  * we just override the previous file.
+     */
+    // Filename of hot update chunks
+    hotUpdateChunkFilename: 'hot/hot-update.js',
+    // The main hot update filename
+    hotUpdateMainFilename: 'hot/hot-update.json'
+  },
+  module: {
+    rules: [
+      {
+        // Apply Babel transformations to JavaScript files only
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: {
+          // Integrate Babel with Webpack
+          loader: 'babel-loader',
+          options: {
+            /**
+             * Presets are Babel plugins.
+             * @babel/preset-env allows us to write next generation JavaScript.
+             * (promises, arrow functions, spread & rest operator...)
+             */
+            presets: ['@babel/preset-env']
+          }
+        }
+      }
+    ]
+  },
+  devServer: {
+    host: 'localhost',
+    port: 3000,
```

Comment:
Can we do 8080, port 3000 is so often already taken for me by Rails app
```suggestion
    port: 8080,
```

Reply:
:+1: